lib/controller/accountscontroller.php

```diff
@@ -252,22 +253,20 @@ public function create($accountName, $emailAddress, $password,
 	 * @param string $cc
 	 * @param string $bcc
 	 * @param int $draftUID
-	 * @param int $messageId
+	 * @param string $messageId
 	 * @param mixed $attachments
 	 * @return JSONResponse
 	 */
 	public function send($accountId, $folderId, $subject, $body, $to, $cc,
 		$bcc, $draftUID, $messageId, $attachments) {
 		$account = $this->accountService->find($this->currentUserId, $accountId);
-		if (!$account instanceof Account) {
-			return new JSONResponse(
-				array('message' => 'Invalid account'),
-				Http::STATUS_BAD_REQUEST
-			);
+		if ($account instanceof UnifiedAccount) {
+			list($account, $folderId, $messageId) = $account->resolve($messageId);
 		}
```

Comment:
Could you please re-add the above ``if (!$account instanceof Account) {`` check like:

```PHP
if ($account instanceof UnifiedAccount) {
    list($account, $folderId, $messageId) = $account->resolve($messageId);
} elseif (!$account instanceof Account) {
   return ....
}
```
Otherwise requests could result into 500 errors